docs/format_signatures.rst

```diff
@@ -0,0 +1,127 @@
+
+.. confval:: html_wrap_signatures_with_css
+
+   Specifies for which `Sphinx domains
+   <https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html>`__
+   this CSS-based formatting is enabled.
+
+   Supported types are:
+
+   :py:obj:`bool`
+       If `True`, enable CSS-based formatting for all domains.  If `False`,
+       disable CSS-based formatting for all domains.
+
+   :py:obj:`list[str]`
+       List of domains for which to enable CSS-based formatting.
+
+   The default value is :python:`True`
+
+.. confval:: html_wrap_signatures_with_css_column_limit
+
+   Maximum signature length before function parameters are displayed on separate
+   lines.
+
+   Only applies to domains for which :confval:`html_wrap_signatures_with_css` is
+   enabled.
+
+   The default value is :python:`68`.
+
+.. rst-example::
+
+   .. py:function:: long_function_signature_example(\
+                      name: int, \
+                      other_param: list[str], \
+                      yet_another: bool = False, \
+                      finally_another: str = "some long string goes here"\
+                    ) -> Tuple[A, Very, Long, Return, Type]
+
+clang-format-based function parameter wrapping
+-----------------------------------------------
+
+There is a more powerful alternative formatting mechanism based on `clang-format
+<https://clang.llvm.org/docs/ClangFormat.html>`__.  This supports C, C++, Java,
+JavaScript, Objective-C, and C#.
+
+This functionality is available as a separate extension included with this
+theme.  To use it, you must include it in your conf.py file and you must also
+set the :confval:`clang_format_signatures_domain_styles` configuration option.
+
+.. code-block:: python
+
+    extensions = [
+        # other extensions...
+        "sphinx_immaterial.format_signatures",
+    ]
+
+    clang_format_signatures_domain_styles = {
+        "cpp": """{
+        BasedOnStyle: LLVM,
+        ColumnLimit: 68,
+        }""",
+    }
+
+.. confval:: clang_format_signatures_domain_styles
+
+   Dictionary that specifies the `clang-format style options
+   <https://clang.llvm.org/docs/ClangFormatStyleOptions.html>`__ for each
+   domain.  Formatting is enabled *only* for domains that are listed.
+
+   .. literalinclude:: conf.py
+      :language: python
+      :start-after: # BEGIN: sphinx_immaterial.format_signatures extension options
+      :end-before: # END: sphinx_immaterial.format_signatures extension options
+
+   When specifying an inline style (as opposed to a predefined style), it is
+   necessary to enclose the style in curly braces, as in the example above.
+   Since the predefined styles (such as ``Google``, ``LLVM``, etc.) do not use a
+   column limit of 68, it is not recommended to use a predefined style.
+
+   .. tip::
+
+      It is recommended to include ``ColumnLimit: 68`` as a style option, as
+      this is the default for the CSS-based wrapping, and is the number of
```

Comment:
```suggestion
      this is the default for the CSS-based wrapping, and it is the number of
```
Conjunctions that follow a comma should involve a subject. This rule of English is a bit hidden here because of the prepositional anecdote (which are also delimited by commas).

Reply:
On second thought, its probably better to break this up into 2 sentences. Sorry, I'm being "that guy", but grammar mistakes always
1. throw me when I encounter them in other people's docs (flask docs simply don't use commas unless it is followed by a conjunction 😠 )
2. may cause difficulty when input to google translate (or similar)

It looks like "as" is used here _as_ a preposition and a conjunction (which is followed by another conjunction "and" => run-on sentence).

Reply:
Fixed.